tests/Builder/Query/TypeOperatorTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Query;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+/**
+ * Test $type query
+ */
+class TypeOperatorTest extends PipelineTestCase
+{
+    public function testQueryingByArrayType(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::match(
+                zipCode: Query::type('array'),
```

Comment:
As an idea for the future: we could provide an enum for the various BSON types supported by this operator here.